Add duros extension

ostempel opened the pull request:
## Description

Extract duros-extension from GEPM and provide extra functionality for QoS types.

<!--
If possible, please reference other issues or pull requests.

Closes #<the-issue-number-to-close>.

References:

- ...

If not already described in a referenced issue, please describe your PR and the motivation behind it. Just try to make life easy for the reviewers.

Please be aware that the pull request's title will become part of the release notes, so try to make it understandable.
-->

<!--
If you would like to add something to the release notes for the next metal-stack release (metal-stack/releases), you can do so by adding SPECIAL SECTIONS (code blocks) in this PR. Please only add a section when this is relevant for the entire project.

You can use the following snippets as an example:

## Release Notes

### Breaking Change

```BREAKING_CHANGE
Description of the breaking change and what an operator needs to do about it.
This section is **not** intended for documentation of internal breaking changes.
Release notes are meant to be read by users and operators of metal-stack, not metal-stack developers.
```

### Required Actions

```ACTIONS_REQUIRED
Description of the required action for operators.
```

### Noteworthy

```NOTEWORTHY
Description of noteworthy release information for the metal-stack project that users or operators should know.
```
-->

ostempel committed:
initial commit

ostempel committed:
merge

ostempel committed:
add configs

ostempel committed:
draft reconcile of duros

ostempel committed:
remove metal-stack dependencies

ostempel committed:
adjust naming add duros crd

ostempel committed:
finish reconcile

ostempel committed:
draft deletion

ostempel committed:
remove partitionId from duros cfg

vknabel commented:
@ostempel any progress on this? Builds are currently failing preventing further reviews.

Gerrit91 committed:
Review.

Gerrit91 committed:
Update go version for builder image.

Gerrit91 committed:
Allow more.

Gerrit91 committed:
Add firewall scheme.

Gerrit91 committed:
Update.

Gerrit91 committed:
Log error.

Gerrit91 committed:
More permissions for secrets.

Gerrit91 committed:
Debug.

Gerrit91 committed:
Read from zones.

Gerrit91 committed:
Add api-cert.

Gerrit91 committed:
S

Gerrit91 committed:
Put MR into shoot namespace.

Gerrit91 committed:
Fix.

mac641 committed:
ci: include generating sbom in docker workflow

mac641 committed:
Merge pull request #3 from metal-stack/sbom

include sbom in container image

Gerrit91 commented on pkg/controller/duros/actuator.go:
```diff
@@ -0,0 +1,624 @@
+		},
+	}
+
+	role := rbacv1.ClusterRole{
```
We should not create cluster wide resources inside the seed, otherwise it conflicts with other shoots.

Gerrit91 committed:
Fix seed rbac for duros controller must be a role, not cluster role.

Gerrit91 committed:
Pass the namespace properly.

Gerrit91 committed:
Fix typo.

Gerrit91 committed:
Refactor: Rename opts and apply duros CRDs a it later to pickup restopts

Gerrit91 committed:
A couple of improvements.

- Fix duros cr missing namespace
- Move global CWNPs out of MR to prevent deletion issues
- Remove concept of zones for now (too complex)

Gerrit91 committed:
Fix.

Gerrit91 committed:
Update RBAC.

Gerrit91 committed:
Add watch.

Gerrit91 committed:
Rename and maintain old name.
